### DISCUSSION TITLE
DecodingOptions

### DISCUSSION THREAD

**Discussion by pop2pop3** [openai/whisper#39] [Show and tell]
Since PyTorch CPU does not assist fp16, so I change DecodingOptions fp16 parameter to False, this will slow down the decoding process since CPU is reasoned through the use of fp16 combined precision calculation. At least this happened on arm64 CPU.

**Comment by Strangemother** [comment]
Thank you - very helpful.

As an example I applied it directly to the transcribe method because I'm lazy:

```
import whisper
model = whisper.load_model('base')
result = model.transcribe("audio.mp3", fp16=False)
print(result["text"])
```

**Reply by pop2pop3** [reply]
Sure, or you can change from `dtype = torch.float16` to `dtype = torch.float32` directly in `transcribe()` function,  so you don't need to add another argument
